contracts/ElectionPractice.sol

```diff
@@ -0,0 +1,166 @@
+pragma solidity ^0.5.0;
+
+/**
+ * @title - How to use this?
+ * @dev - deploy contract to DEXON netowkr
+ * @dev - copy deployed address to https://dexon-workshop-vote.netlify.com/
+*/
+
+/**
+ * @title - In this Election Contract, it could take as many rounds of election.
+ *        - As a user of this contract, you are able to
+ *        - 1. Register as a candiate of the election.
+ *        - 2. One vote the candiate of your choose, and can fund any of the candiate as much money as you want.
+ *        - 3. Candiate will get the fund if the round is over.
+ *
+ * @dev - There would be only 1 active round each at a time.
+ *      - Before round start, any user can use their `wallet` to register as candiate "once".
+ *      - Only the contract owner can start voting process and call a voting process to the end.
+ *      - Once a voting proces start, no more user can register as candiate.
+ * 
+ * @notice - Search for "TODO" for what you need to implement.
+ *      
+ *
+*/
+
+/**
+ * @title - Events
+ * @dev - Events will serve two purposes
+ *      - 1. Update Election UI via websocket
+ *      - 2. To save data that you want on the blockchain but won't be needed in the contract
+ * 
+ * @notice - do not change the name of event, otherwise the UI will not update automatically
+*/
+contract Events {
+    event voteStart(uint round);
+    event elected(uint round, address candidate, string name, uint vote);
+    event reset(uint round);
+    event registered(uint round, address candidate);
+    event voteCandidate(uint round, address candidate, address voter);
+    event sponsorCandidate(uint round, address candidate, string name, address sponsor, uint amount);
+    event refund(address candidate, string name, uint amount, uint round);
+} 
+ 
+contract Election is Events {
+
+    /**
+     * @notice - This is for functions that can be used/called by the owner.
+     */
+    address public owner;
+    
+    constructor() public {
+        // TODO - Set up owner right after the contract get deployed.
+    }
+    
+    /**
+     * @title isVoting
+     * @dev a mark to determine if 
+     * @notice a public variable that is used for contract and UI update
+     */
+    bool public isVoting = false;
+    
+    function startVoting() public {
+        /**
+         * TODO - to make voting process start
+         * TODO - only contract owner can call this function
+         * TODO - emit event voteStart
+        */
+    }
+    
+
+    /**
+     * @notice - With solidity, it gives you "mapping" as one of the data store mechanism.
```

Comment:
*  - It's a key-value pair
* - However, you cannot iterate the keys

Reply:
ok